pkgs/development/interpreters/ruby/parse_version.nix

```diff
@@ -0,0 +1,91 @@
+# Contains the ruby version heuristics
+{ lib }:
+with lib;
+let
+  parseVersion = versionStr:
+    let
+      pair = splitString "-" versionStr;
```

Comment:
Well, Nix is not a general purpose language and it's definitely not ideal for parsing stuff. This kind of thing tends to slow down operations like `nix-env -qa` and lead to memory bloat.

BTW, it should be `parse-version.nix`, not `parse_version.nix`.